RHELPLAN-56586 - sync collections related changes from template to network role

nhosoi opened the pull request:
- Adding .travis/runcollection.sh.
- Enabling collection as part of tox/travis tests.

ffmancera commented on .travis/runcollection.sh:
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
```
This license statement is for MIT license but AFAICS in https://github.com/linux-system-roles/network/blob/main/LICENSE the license is BSD-3. Is this intentional?

richm replied:
This file is copied from https://github.com/linux-system-roles/template/blob/master/.travis/runcollection.sh where it is MIT licensed.   I suppose it doesn't matter.  Hopefully all of these copied files will go away soon . . . https://github.com/linux-system-roles/tox-lsr - so use whatever license you want

nhosoi replied:
runcollection.sh is one of the tox test script originally checked in linux-system-roles/template repo. The test scripts are under MIT. It is consistent with these files in the network repo.
```
ansible_pytest_extra_requirements.txt:# SPDX-License-Identifier: MIT
custom_requirements.txt:# SPDX-License-Identifier: MIT
molecule_extra_requirements.txt:# SPDX-License-Identifier: MIT
pylint_extra_requirements.txt:# SPDX-License-Identifier: MIT
pylintrc:# SPDX-License-Identifier: MIT
pytest_extra_requirements.txt:# SPDX-License-Identifier: MIT
tox.ini:# SPDX-License-Identifier: MIT
.travis.yml:# SPDX-License-Identifier: MIT
.yamllint_defaults.yml:# SPDX-License-Identifier: MIT
.yamllint.yml:# SPDX-License-Identifier: MIT
.travis/config.sh:# SPDX-License-Identifier: MIT
.travis/custom.sh:# SPDX-License-Identifier: MIT
.travis/runblack.sh:# SPDX-License-Identifier: MIT
.travis/runcoveralls.sh:# SPDX-License-Identifier: MIT
.travis/runflake8.sh:# SPDX-License-Identifier: MIT
.travis/runpylint.sh:# SPDX-License-Identifier: MIT
.travis/runpytest.sh:# SPDX-License-Identifier: MIT
.travis/runsyspycmd.sh:# SPDX-License-Identifier: MIT
.travis/utils.sh:# SPDX-License-Identifier: MIT
```

ffmancera replied:
Oh I didn't know where this was coming from. My bad on this. Thanks for explaining!

ffmancera commented on .travis/runcollection.sh:
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
I think you can do ` #!/bin/bash -e` instead of using `set -e` later.

richm replied:
sure - if you want, please propose a change to the source script at https://github.com/linux-system-roles/template/blob/master/.travis/runcollection.sh then we can again copy it to every other system role repo

ffmancera commented on .travis/utils.sh:
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
 # SPDX-License-Identifier: MIT
```
This not new added, but why the license is MIT? Any hint @tyll?

richm replied:
@nhosoi please remove this file from the commit if it has no changes

nhosoi replied:
??? All the changes made on utils.sh or just this `# SPDX-License-Identifier: MIT` line? If former, I had to fix some shellcheck errors in the file.

richm replied:
Since we have established that files copied from `template/` can retain their MIT license even though the rest of `network/` is licensed under BSD-3, then we don't need to change the license in this file.
And hopefully this will all go away soon . . . https://github.com/linux-system-roles/tox-lsr

nhosoi replied:
Anyway, I reverted utils.sh since the changes are not needed as we skip shellcheck for network.

ffmancera commented on tox.ini:
```diff
@@ -108,10 +108,12 @@ commands =
     docker{env:LSR_MOLECULE_DOCKER_VERSION:}
     molecule<3
     selinux
+    wcwidth==0.1.9;python_version=="3.6"
```
What happens if the python version is lower?

richm replied:
not sure - @nhosoi do we need this change for collections?  If not, please remove this change from the commit

nhosoi replied:
No, not for collections. sync-template brought it in. All right. Removing the line.

ffmancera commented on tox.ini:
```diff
@@ -176,7 +188,7 @@ addopts = -rxs
 show_source = true
 max-line-length = 88
-ignore = E402,W503
+ignore = E402,W503,E501
```
Why should we ignore E501?

nhosoi replied:
If you convert a python library in your role to the collections format, the import statement could be longer than the original one. For example, `from ansible.module_utils.network_lsr.utils import Util` becomes 
`from ansible_collections.fedora.system_roles.plugins.module_utils.network_lsr.utils import Util`. To access an object in the collections, it's supposed to give Fully Qualified Collection Name (FQCN), which starts with ansible_collections.collection_namespace.collection_name. Plus, `module_utils` dir is moved to `plugins` in the collections format. To handle the length, we had to add E501.

tyll replied:
Good catch @ffmancera - @nhosoi instead of disabling the error globally, it should be disabled for the respective line with appending a comment like `# noqa: E501` to ensure that the remaining files are still checked.

ffmancera reviewed (changes requested):
Hello! It looks quite good, I made some comments below. Please, could you add more details in the commit description? Thanks!

nhosoi commented:
> Hello! It looks quite good, I made some comments below. Please, could you add more details in the commit description? Thanks!

Thank you for your reviews, @ffmancera, @richm and @tyll. I've pushed the 2nd commit to fix the issues pointed by all of you. Could you please review the patch one more time? Then, I'm going to squash them into one with more commit description. Thanks.

ffmancera reviewed (approved):
LGTM, thank you!

nhosoi commented:
> LGTM, thank you!

Thanks a lot, @ffmancera! Squashed into one commit and added more detailed info to the [commit comment](https://github.com/linux-system-roles/network/pull/296/commits/025d007a907a0c70fcf5eb824c20c824721ba274). Hope it makes sense...

tyll commented:
@cathay4t please review and merge if you agree. Thanks.

cathay4t commented:
@nhosoi Thanks for the pull request. Could you rebase it and we can merge it?

nhosoi commented:
> @nhosoi Thanks for the pull request. Could you rebase it for us to merge it?

Thank you, @cathay4t. Done.
